isminet/clients/unifi.py

```
@@ -0,0 +1,426 @@
+            )
+
+        data = response["data"]
+        if not isinstance(data, list):
+            raise ResponseValidationError(
+                "Invalid response structure: 'data' field must be a list", None
+            )
+
+        try:
+            return [model_class(**item) for item in data]
+        except ValidationError as e:
+            raise ResponseValidationError("Failed to validate response items", e)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)